crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3212,15 +3221,25 @@ pub async fn set_default_payment_method(
     key_store: domain::MerchantKeyStore,
     customer_id: &str,
     payment_method_id: String,
+    storage_scheme : Option<MerchantStorageScheme>,
 ) -> errors::RouterResponse<CustomerDefaultPaymentMethodResponse> {
     //check for the customer
+    let storage_scheme = match storage_scheme {
+        Some(v) => v,
+        None => {
+            let account = db.find_merchant_account_by_merchant_id(&merchant_id, &key_store)
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)?;
+            account.storage_scheme
+        }
+    };
```

Comment:
Why is the redundant DB call `find_merchant_account` needed?

Reply:
in case calling function doesnt have merchant account, we need storage scheme

Reply:
No we have merchant_id getting passed here anyway cant we instead pass `merchant_account` here and use it.

Reply:
sure